Classes/common/query/CDTQQuerySqlTranslator.m

```diff
@@ -299,6 +335,20 @@ + (NSString *)chooseIndexForFields:(NSSet *)neededFields fromIndexes:(NSDictiona
     return chosenIndex;
 }
 
++ (NSString *)getTextIndexFromIndexes:(NSDictionary *)indexes
+{
+    NSString *textIndex = nil;
+    for (NSString *indexName in [indexes allKeys]) {
+        NSDictionary *indexDefinition = indexes[indexName];
+        NSString *indexType = indexDefinition[@"type"];
```

Comment:
I'd probably use a single line here as you have elsewhere, `NSString *type = indexes[indexName][@"type"]`.